plugins/out_stackdriver/stackdriver.c

```diff
@@ -33,6 +34,65 @@
 #include <mbedtls/base64.h>
 #include <mbedtls/sha256.h>
 
+/* Validate resource labels against required labels for a monitored resource */
+static int validate_resource_labels(struct flb_stackdriver *ctx)
+{
+    int i;
+    int m_cnt = 0;
+    int ret = 0;
+
+    for (i = 0; i < ctx->resource->label_cnt; ++i) {
+        flb_plg_debug(ctx->ins,
+            "Validating label %s", ctx->resource->labels[i].label);
+            if (ctx->resource->labels[i].value == NULL) {
+                if (strncmp(ctx->resource->labels[i].label, "project_id", 10) == 0 &&
+                    ctx->project_id != NULL) {
+                    ctx->resource->labels[i].value = flb_sds_create(ctx->project_id);
+                    ++m_cnt;
+                }
+                else if (ctx->resource->metadata_enabled) {
+                    flb_plg_debug(ctx->ins, "Metadata server enabled");
+                    if (strncmp(ctx->resource->labels[i].label, "project_id", 10) == 0) {
```

Comment:
This should be `11` to include the trailing `'\0'` and ensure an exact match (and prevent, e.g., a label named `project_identity`). Ditto for the other labels.

Reply:
Oh yeah, overlooked the null termination. Will fix.